libres/lib/config/config_schema_item.cpp

```diff
@@ -66,398 +66,465 @@ typedef struct validate_struct validate_type;
+                                               value));
+                    break;
+                case (CONFIG_INT):
+                    if (!util_sscanf_int(value, NULL))
+                        config_error_add(
+                            error_list,
+                            util_alloc_sprintf(
+                                "Failed to parse:%s as an integer.", value));
+                    break;
+                case (CONFIG_FLOAT):
+                    if (!util_sscanf_double(value, NULL)) {
+                        config_error_add(
+                            error_list,
+                            util_alloc_sprintf("Failed to parse:%s as a "
+                                               "floating point number.",
+                                               value));
+                        OK = false;
+                    }
+                    break;
+                case (CONFIG_PATH):
+                    // As long as we do not reuqire the path to exist it is just a string.
+                    break;
+                case (CONFIG_EXISTING_PATH): {
+                    char *path = config_path_elm_alloc_abspath(path_elm, value);
+                    if (!util_entry_exists(path)) {
+                        config_error_add(
+                            error_list,
+                            util_alloc_sprintf(
+                                "Can not find entry %s in %s ", value,
+                                config_path_elm_get_relpath(path_elm)));
+                        OK = false;
+                    }
+                    free(path);
+                } break;
+                case (CONFIG_EXECUTABLE): {
+                    if (!util_is_abs_path(value)) {
+
+                        char *relocated = alloc_relocated__(path_elm, value);
+                        if (util_file_exists(relocated)) {
+                            if (util_is_executable(relocated))
+                                stringlist_iset_copy(token_list, iarg,
+                                                     relocated);
+
+                            free(relocated);
+                            break;
+                        }
+
+                        free(relocated);
+
+                        if (strstr(value, "/")) {
+                            /*
                    * value not naked name, which means it was relative path
                    * that wasn't found
                    */
 
-                config_error_add( error_list , util_alloc_sprintf("Could not locate executable:%s ", value));
-                break;
-              }
+                            config_error_add(
+                                error_list,
+                                util_alloc_sprintf(
+                                    "Could not locate executable:%s ", value));
+                            break;
+                        }
 
-              /*
+                        /*
```

Comment:
Indentation of comment now looks weird because the indentation of the code has changed.